exporter/elasticsearchexporter/README.md

```diff
@@ -51,8 +51,10 @@ This exporter supports sending OpenTelemetry logs and traces to [Elasticsearch](
   - `date_format`(default=`%Y.%m.%d`): Time format (based on strftime) to generate the second part of the Index name.
 - `pipeline` (optional): Optional [Ingest pipeline](https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html) ID used for processing documents published by the exporter.
 - `flush`: Event bulk indexer buffer flush settings
-  - `bytes` (default=5000000): Write buffer flush size limit.
-  - `interval` (default=30s): Write buffer flush time limit.
+  - `bytes` (DEPRECATED, use `flush.min_documents` instead): Write buffer flush size limit. WARNING: This configuration is ignored.
+  - `min_documents` (default=125): Minimum number of documents in the buffer to trigger a flush immediately.
+  - `max_documents` (default=0): Maximum number of documents in a request. In practice, the number of documents in a request may exceed MaxDocuments if the request cannot be split into smaller ones.
+  - `interval` (default=30s): Max age of a document in the buffer. A flush will happen regardless of the size of content in buffer.
```

Comment:
Thanks for the pointer, it is the first time I see exporterbatcher being used. I briefly thought about it but forgot about it. I think it is a good practice to align on config options with other exporters to make it more user friendly. Moving to `exporterbatcher.Config` is a good idea.

Some points to think about:
- In the config, `min_size_items` uses request `ItemsCount`. The unit of it is the number of flattened spans, data points or log records. Note that metrics `ItemsCount` is the number of data points, not metrics, such that it does not have 1:1 correspondence like logs and traces. In the future, when we support metrics, we'll have to decide on the unit of min_size_items, whether it is ES documents or flattened p-data count (i.e. spans, data points or log records).

Reply:
> In the config, min_size_items uses request ItemsCount. The unit of it is the number of flattened spans, data points or log records. Note that metrics ItemsCount is the number of data points, not metrics, such that it does not have 1:1 correspondence like logs and traces. In the future, when we support metrics, we'll have to decide on the unit of min_size_items, whether it is ES documents or flattened p-data count (i.e. spans, data points or log records).

Even if the item count were the number of metrics, it still wouldn't necessarily match the number of documents we will end up writing. (Actually I think it would be more wrong.)

If we implement things [as we did in apm-data](https://github.com/elastic/apm-data/blob/main/input/otlp/metrics.go), then if every data point had distinct dimensions there would be a 1:1 correspondence. We'll only have fewer due to grouping by dimensions (and timestamps, units, etc.)

I think in the ideal world we would leave that to Elasticsearch, and write one "document" per data point. Then Elasticsearch would aggregate them for storage and query efficiency according to the data stream definition.

Reply:
At this point I wonder if it makes sense to just remove support of `flush.*` outright so that collector fails at config parsing time, rather than to ignore deprecated config and operate with new defaults under `batcher.*`. To me it is more user-friendly to fail rather than to log and expect the user to notice it.

Reply:
If it's possible to deprecate and translate old to new, then remove later, I'd do that. Otherwise sounds good.

Reply:
Added a `handleDeprecations` for a mix of them. `flush.bytes` cannot be translated, and will return an error if configured. On the other hand, `flush.interval` will be translated to `batcher.flush_timeout` with a WARN log.

Reply:
@axw here comes the metrics support: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33513/files#diff-fb57c737508bb9f781260c25e0471d0bd33af0306a7267f3ec903427bf0aeb7cR189

It is doing grouping, like what we mentioned above. Given that min_size_items is counting data points, it will not be 1:1 to document. I don't think it is a blocker, but the defaults will be slightly suboptimal for metrics compared to logs and traces if we move forward with this. There's always an escape hatch to specify a different min_size_items for metrics only.

Reply:
:+1: I guess we could also do the counting of data points in the same way, i.e. count hashed timestamp & attributes. I think we could revise this as we go?